src/v/http/probe.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include <boost/beast/http.hpp>
+
+#include <cstdint>
+
+namespace http {
+
+struct client_probe {
+    using verb = boost::beast::http::verb;
```

Comment:
should the client probe give you a struct instead

```
auto p = probe.GET()
auto p = probe.PUT()
....

that you can capture for the life of the request at the handler level. without importing the boost beast headers. 
``

Reply:
This will move switch statement to the http client. I just removed the dependency on boost here.